### PR TITLE
Fix classification report if dataset has no labels

### PR DESCRIPTION
**Pull request by alanakbik**
If the test dataset has no labels, and the model makes no predictions, no evaluation numbers (precision/recall/F1) can be computed. This causes errors in cases such as this: 

```python
import flair
from flair.data import Corpus
from flair.datasets import ColumnCorpus, WNUT_17
from flair.embeddings import TransformerWordEmbeddings
from flair.models import SpanClassifier
from flair.trainers import ModelTrainer

flair.set_seed(123)

corpus: Corpus = WNUT_17().downsample(0.01)

label_dictionary = corpus.make_label_dictionary("ner", add_unk=True)

embeddings = TransformerWordEmbeddings(
    model="distilbert-base-uncased", layers="-1", subtoken_pooling="first", fine_tune=True, use_context=True
)

tagger = SpanClassifier(embeddings=embeddings, label_dictionary=label_dictionary)

trainer = ModelTrainer(tagger, corpus)

trainer.fine_tune(
    "./results/",
    max_epochs=1,
)
```

This PR slightly refactors the `evaluate()` code in the `DefaultClassifier` such that the error handling for this case becomes more explicit. This also fixes the above error.

### CODE REVIEW

File: flair/nn/model.py [hunk 1]
```diff
@@ -434,71 +434,66 @@ def evaluate(
...
-        result = Result(
-            main_score=main_score,
-            detailed_results=detailed_result,
-            classification_report=classification_report_dict,
-            scores=scores,
-        )
+        else:
+            # issue error and default all evaluation numbers to 0.
+            error_text = (
+                f"ACHTUNG! It was not possible to compute evaluation values because: "
```
Comment (helpmefindaname) [inline flair/nn/model.py:484]: I think this PR is a good opportunity to remove the German word and phrase it completely in english.
Besides that:
The current output looks like this:
```
2023-11-13 11:07:43,133 ACHTUNG! It was not possible to compute evaluation values because: - The evaluation data has no gold labels for label_type='nel'!- And no predictions were made!Double check your corpus (if the test split has labels), and how you initialize the ModelTrainer!
2023-11-13 11:07:43,133 ACHTUNG! It was not possible to compute evaluation values because: - The evaluation data has no gold labels for label_type='nel'!- And no predictions were made!Double check your corpus (if the test split has labels), and how you initialize the ModelTrainer!
```
I would
* Fix the spacing
* Not log it twice: either log the error or add it in the detailed_results, but not both

**Review by helpmefindaname — CHANGES_REQUESTED** [review]

File: flair/nn/model.py (hunk 1 above)
Comment (alanakbik) [reply flair/nn/model.py]: Fixed in latest push.
